mlflow/store/model_registry/databricks_workspace_model_registry_rest_store.py

```diff
@@ -13,6 +15,49 @@
 from mlflow.utils.logging_utils import eprint
 from mlflow.utils.uri import _DATABRICKS_UNITY_CATALOG_SCHEME
 
+_logger = logging.getLogger(__name__)
+
+
+def _extract_workspace_id_from_run_link(run_link: str) -> str | None:
+    """
+    Extract workspace ID from a Databricks run link URL.
+
+    Args:
+        run_link: URL like "https://workspace.databricks.com/?o=10002#mlflow/experiments/test-exp-id/runs/runid"
+        The workspace ID is the part after the ?o= in the URL, and before the #,
+        e.g. 10002 in the example above.
+        The run_link is only present if the run was logged in a Databricks Workspace
+        different from the registry workspace.
+
+    Returns:
+        The workspace ID as a string, or None if not found or invalid
+    """
+    if not run_link:
+        return None
+
+    warning_msg = (
+        "Unable to get model version source run's workspace ID from source run link. "
+        "The source workspace ID on the destination model version will be set to the "
+        "registry workspace ID."
+    )
+
+    try:
+        parsed_url = urlparse(run_link)
+        query_params = parse_qs(parsed_url.query)
+        workspace_id_params = query_params.get("o")
+        if not workspace_id_params:
+            _logger.warning(warning_msg)
+            return None
+        workspace_id = workspace_id_params[0]
+        workspace_id_int = int(workspace_id)
+        if workspace_id_int < 0:
```

Comment:
```suggestion
        if int(workspace_id) < 0:
```

let's remove useless assignment.